bin/worker.js

```diff
@@ -176,41 +172,35 @@ async function main () {
   // level docker-worker components.
   config.docker = require('../lib/docker')();
 
-  // Wrapped stats helper to support generators, etc...
-  config.stats = new Stats(config);
-  config.stats.record('workerStart', Date.now()-os.uptime() * 1000);
-  debug('passed stats')
-
-  config.queue = new taskcluster.Queue({
+  let monitor = await base.monitor({
+    project: 'docker-worker',
     credentials: config.taskcluster,
-    stats: base.stats.createAPIClientStatsHandler({
-      drain: config.stats.influx,
-      tags: {
-        component: 'docker-worker',
-        workerId: config.workerId,
-        workerGroup: config.workerGroup,
-        workerType: config.workerType,
-        instanceType: config.workerNodeType,
-        provisionerId: config.provisionerId
-      }
-    })
+    mock: profile === 'test'
   });
 
-  config.validator = await base.validator();
-  config.validator.register(require('../schemas/payload'));
+  config.monitor = monitor.prefix(
```

Comment:
I should probably make `monitor.prefix()` take an array of prefixes and join them internally to allow you to avoid concat-ing everything yourself. Not a bug on this review, just a note to myself.